addons/web_editor/static/lib/odoo-editor/test/spec/color.test.js

```diff
@@ -39,4 +39,22 @@ describe('applyColor', () => {
             contentAfter: '<p>ab<font style="background-color: rgb(255, 0, 0);">[]\u200B</font>cd</p>',
         });
     });
+    it('should apply a color on empty selection', async () => {
+        await testEditor(BasicEditor, {
+            contentBefore: '<p>[<br></p><p><br></p><p>]<br></p>',
+            stepFunction: setColor('rgb(255, 0, 0)', 'color'),
+            contentAfter: '<p><font style="color: rgb(255, 0, 0);">[<br></font></p>' +
+                          '<p><font style="color: rgb(255, 0, 0);"><br></font></p>' +
+                          '<p>]<br></p>',
+        });
+    });
+    it('should apply a background color on empty selection', async () => {
+        await testEditor(BasicEditor, {
+            contentBefore: '<p>[<br></p><p><br></p><p>]<br></p>',
+            stepFunction: setColor('rgb(255, 0, 0)', 'background-color'),
+            contentAfter: '<p><font style="background-color: rgb(255, 0, 0);">[<br></font></p>' +
+                          '<p><font style="background-color: rgb(255, 0, 0);"><br></font></p>' +
+                          '<p>]<br></p>',
```

Comment:
Same for this one, with color changed to background-color, obviously.

Reply:
@dhba-odoo You can use `contentAfterEdit` to test with the zws still present and `contentAfter` to test the result after the clean. If you search for "contentAfterEdit" you will see lots of tests use this.

Reply:
@dmo-odoo done!

Reply:
@dhba-odoo Thank you ! Tomorrow I didn't have time to reply properly to your message above so I only replied to the `contentAfter` part so that you would not be blocked by it anymore. Now I'm replying to your point 3:

I understand that `getSelectedNodes` is not going to return the third P. However, In this case, I think it might work to do so something like this:
```
const targetedNodes = Set(getSelectedNodes(editor.editable));
if (isEmptyBlock(range.endContainer)) {
    targetedNodes.add(range.endContainer);
}
```

cc @Zinston Looks like a second place where a `getTargetedNodes` util might be useful. If we ever get a third one we'll need to create a proper util for it.

Reply:
@dmo-odoo Applied suggested changes. Please look into it.

Reply:
@dhba-odoo Awesome, thanks ! Is there a reason why you also add `...descendants(range.endContainer)` ? Because if they were selected to begin with then they would have been returned by `getSelectedNodes`. If they weren't then we only want the add the parent.

Reply:
@dmo-odoo 
Selecting an empty line as a selection or the last node of the selection is empty at that time the endContainer value will be the p tag.
https://github.com/odoo/odoo/blob/15.0/addons/web_editor/static/lib/odoo-editor/src/commands/commands.js#L566
This line prevents applying color on block level tag and because the p tag is a block tag the color will be not applied to it.
if we want to apply color on the last p tag when it's empty we need to add its child nodes into the list which is br.
so, to apply color on the last empty tag I have added its descendants in my code.

And yes, getSelectedNodes is not returning it.

are you expecting the p tag wrapped inside the font tag? if yes then we need to change the functionality of how color applies because currently, we are applying the color on child nodes of p tag and not p tag.